federation/pkg/federation-controller/sync/controller.go

```diff
@@ -416,37 +460,65 @@ func syncToClusters(clustersAccessor clustersAccessorFunc, operationsAccessor op
 		return redeliverForFailure
 	}
 
-	return reconciled
+	return statusOK
 }
 
 type clusterObjectAccessorFunc func(clusterName string) (interface{}, bool, error)
 
 // clusterOperations returns the list of operations needed to synchronize the state of the given object to the provided clusters
-func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, clusters []*federationapi.Cluster, obj pkgruntime.Object, accessor clusterObjectAccessorFunc) ([]util.FederatedOperation, error) {
-	key := federatedtypes.ObjectKey(adapter, obj)
-	operations := make([]util.FederatedOperation, 0)
+func clusterOperations(adapter federatedtypes.FederatedTypeAdapter, clusters []*federationapi.Cluster, fedObj pkgruntime.Object, accessor clusterObjectAccessorFunc) ([]util.FederatedOperation, error) {
+	key := federatedtypes.ObjectKey(adapter, fedObj)
+	currentObjs := make(map[string]pkgruntime.Object)
 	for _, cluster := range clusters {
 		clusterObj, found, err := accessor(cluster.Name)
 		if err != nil {
 			return nil, fmt.Errorf("Failed to get %s %q from cluster %q: %v", adapter.Kind(), key, cluster.Name, err)
 		}
-		// The data should not be modified.
-		desiredObj := adapter.Copy(obj)
 
-		var operationType util.FederatedOperationType = ""
 		if found {
-			clusterObj := clusterObj.(pkgruntime.Object)
-			if !adapter.Equivalent(desiredObj, clusterObj) {
+			currentObjs[cluster.Name] = clusterObj.(pkgruntime.Object)
+		} else {
+			currentObjs[cluster.Name] = nil
+		}
+	}
+
+	scheduledObjs := make(map[string]pkgruntime.Object)
+	if schedAdapter, ok := adapter.(federatedtypes.SchedulerAdapter); ok {
+		var err error
+		scheduledObjs, err = schedAdapter.Schedule(fedObj, currentObjs)
+		if err != nil {
+			return nil, fmt.Errorf("Failure executing Schedule Hook on %s %q: %v", adapter.Kind(), key, err)
+		}
+	} else {
+		for clusterKey := range currentObjs {
+			// The same object propagated into all clusters.
+			desiredObj := adapter.Copy(fedObj)
+			scheduledObjs[clusterKey] = desiredObj
+		}
+
+	}
+
+	operations := make([]util.FederatedOperation, 0)
+	for clusterName, currentObj := range currentObjs {
```

Comment:
There is a case where I would want a local HPA to become inactive in a given cluster. One approach is to make the max and min zero, but, but the current local hpa implementation does not agree with replicas to be zero. The other option then is to remove the object all together from the cluster where we dont need the autoscaling to happen (Still contemplating this). 
Will do, but probably as follow up PR. Will raise one for the controller very soon. Will need your review there.. :-)